### PR TITLE
Add iOS to Travis testing

### PR DESCRIPTION
**Pull request by noloader**
This PR adds iOS to Travis testing.

Some of the Android gear was tweaked, but it was minor. The minor tweaks include changing the `AOSP_` prefix to `ANDROID_` prefix for consistency with `IOS_`.

This cut-in of iOS is more comprehensive than Unbound's (at the moment). This cut-in includes different flags tailored for current SDKs, and includes Simulator testing. We will loop back to Unbound shortly.

### CODE REVIEW

File: contrib/ios/setenv_ios.sh [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+#####         Environment          #####
+########################################
+
+# The flags below were tested with Xcode 8 on Travis. If
+# you use downlevel versions of Xcode, then you can push
+# xxx-version-min=n lower. For example, Xcode 6 can use
+# -miphoneos-version-min=5.
+
+# iPhones can be either 32-bit or 64-bit
+if [[ "$IOS_SDK" == "iPhoneOS" && "$IOS_CPU" == "armv7"* ]]; then
```
Comment (wcawijngaards) [inline contrib/ios/setenv_ios.sh]: The `*` looks wrong, in `"armv7"*`.  It is not a regexp match.  Outside of quotes that could filename expand.

File: contrib/ios/setenv_ios.sh [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+if [[ "$IOS_SDK" == "TV" || "$IOS_SDK" == "AppleTV" ]]; then
+    IOS_SDK=AppleTVOS
+fi
+
+if [[ "$IOS_SDK" == "Watch" || "$IOS_SDK" == "AppleWatch" ]]; then
+    IOS_SDK=WatchOS
+fi
+
+if [[ "$IOS_CPU" == "aarch64" || "$IOS_CPU" == "armv8"* ]] ; then
```
Comment (wcawijngaards) [inline contrib/ios/setenv_ios.sh]: And here too `*` in `"armv8"*` looks wrong.  It is not a regexp, and likely filename expansion.  If you want a regexp match you can make a case statement, and have `armv8*))` as one of the options.

File: contrib/ios/setenv_ios.sh (hunk 1 above)
Comment (wcawijngaards) [reply contrib/ios/setenv_ios.sh]: Actually I seem to be wrong, inside `[[ ]]` that is a regex match, sorry for the noise.

File: contrib/ios/setenv_ios.sh (hunk 2 above)
Comment (wcawijngaards) [reply contrib/ios/setenv_ios.sh]: And same here, that seems to be fine, when I tested it. Sorry for the noise.

Comment (noloader) [reply contrib/ios/setenv_ios.sh]: No problems @wcawijngaards. I actually had to ask the Bask folks about it yesterday. See [Not matching string with wildcard in if statement](https://lists.gnu.org/archive/html/help-bash/2020-03/msg00008.html) on the Bash users mailing list.

**Comment by noloader** [comment]
@wcawijngaards,

It looks like `make install` is failing for Aarc64. See [Travis job 660511306](https://travis-ci.org/NLnetLabs/ldns/jobs/660511306). I am seeing that failure intermittently, too.

Master has the fix for [Issue 40](https://github.com/NLnetLabs/ldns/issues/40) applied, so I am not sure what is going on. Maybe Autotools is old on that machine? Do you have any ideas?

I guess you can add Aarch64 to `allow_failures` until we figure out the issue.

**Comment by wcawijngaards** [comment]
The fix in https://github.com/NLnetLabs/ldns/commit/69e900d1c3d0ac05a44fc75b358d7387d890f948 should likely fix that.

**Comment by noloader** [comment]
@wcawijngaards,

I just rebased to fix the Change log conflict.

**Comment by wcawijngaards** [comment]
Oh yeah, sorry about the Changelog in the commit, but packagers that need libldns.pc are likely to need a note that a change has happened to the Makefile rule.  Maybe the travis works now, thanks for the rebase.

**Comment by noloader** [comment]
@wcawijngaards ,

No problems. The fault was mine. I should have been using the latest LDNS sources to make it easy on you :)

**Comment by noloader** [comment]
@wcawijngaards ,

Hold tight... Somehow how fetch and merge went sideways. It undid your changes.

Let me try again.

**Comment by noloader** [comment]
@wcawijngaards,

I went through the process again, and your changes to `Changelog` and `Makefile.in` are being discarded:

```
$ git fetch upstream develop
From https://github.com/NLnetLabs/ldns
 * branch              develop    -> FETCH_HEAD
$ git checkout ios
Already on 'ios'
$ git merge upstream/develop
Already up to date.
$ git reset --soft upstream/develop
$ git commit -S -am "Add iOS to Travis testing"
Please enter the passphrase to unlock the OpenPGP secret key:
...
[ios 20e609b6] Add iOS to Travis testing
 15 files changed, 640 insertions(+), 56 deletions(-)
 create mode 100644 contrib/ios/15-ios.conf
 create mode 100755 contrib/ios/bootstrap_ldns.sh
 create mode 100755 contrib/ios/install_openssl.sh
 create mode 100755 contrib/ios/install_tools.sh
 create mode 100644 contrib/ios/openssl.patch
 create mode 100755 contrib/ios/setenv_ios.sh
$ git push -f
Counting objects: 20, done.
Delta compression using up to 12 threads.
Compressing objects: 100% (20/20), done.
Writing objects: 100% (20/20), 6.81 KiB | 6.81 MiB/s, done.
Total 20 (delta 14), reused 0 (delta 0)
remote: Resolving deltas: 100% (14/14), completed with 13 local objects.
To ssh://github.com/noloader/ldns
 + 4c057871...20e609b6 ios -> ios (forced update)
```

Do you have any idea what I am doing wrong?

**Comment by noloader** [comment]
@wcawijngaards,

I'm not sure what changed (or how I picked up the change), but I had to re-check them out:

```
$ git checkout  upstream/develop -- configure.ac
$ git checkout  upstream/develop -- Makefile.in
$ git checkout  upstream/develop -- Changelog
```

Sorry about that.

**Comment by noloader** [comment]
@wcawijngaards,

I think you got the wrinkles ironed out. The LDNS builds are passing as expected in Travis.
